internal/biz/model/v1beta2/representation_reference.go

```diff
@@ -0,0 +1,63 @@
+package v1beta2
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+	"gorm.io/gorm/schema"
+)
+
+type RepresentationReference struct {
+	ResourceID uuid.UUID `gorm:"type:uuid;column:resource_id;index:unique_rep_ref_idx,unique;not null;constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
+
+	LocalResourceID       string `gorm:"column:local_resource_id;"`
+	ReporterType          string `gorm:"column:reporter_type;index:unique_rep_ref_idx,unique"`
+	ResourceType          string `gorm:"column:resource_type;index:unique_rep_ref_idx,unique"`
+	ReporterInstanceID    string `gorm:"column:reporter_instance_id;index:unique_rep_ref_idx,unique"`
+	RepresentationVersion int    `gorm:"column:representation_version;index:unique_rep_ref_idx,unique"`
+	Generation            int    `gorm:"column:generation;index:unique_rep_ref_idx,unique"`
+	Tombstone             bool   `gorm:"column:tombstone"`
+}
+
+func (RepresentationReference) TableName() string {
+	return "representation_references"
+}
+
+func (r *RepresentationReference) GormDbAfterMigration(db *gorm.DB, s *schema.Schema) error {
+	switch db.Name() {
+	case "sqlite":
+		// SQLite doesn't support covering indexes in the same way
+		break
+	case "postgres":
+		// Covering index for query patterns on local_resource_id, reporter_type, resource_type, reporter_instance_id
+		err := db.Exec(`
+			CREATE INDEX IF NOT EXISTS rep_ref_query_covering_idx 
+			ON representation_references (
+				local_resource_id,
+				reporter_type,
+				resource_type,
+				reporter_instance_id
+			)
+			INCLUDE (resource_id);
+		`).Error
+		if err != nil {
+			return fmt.Errorf("failed to create rep_ref_query_covering_idx: %w", err)
+		}
+
+		// Covering index for resource_id queries with commonly accessed columns included
+		err = db.Exec(`
+			CREATE INDEX IF NOT EXISTS rep_ref_resource_id_covering_idx
+			ON representation_references (resource_id)
+			INCLUDE (
+				reporter_type,
+				representation_version,
+				generation,
+				tombstone);
+		`).Error
+		if err != nil {
+			return fmt.Errorf("failed to create rep_ref_resource_id_covering_idx: %w", err)
+		}
```

Comment:
We should avoid doing manual migrations like this while having gorm automigrations enabled. I believe every time we roll out this code, gorm automigrate will remove the indexes and then this code will re-add them, and we don't want that running on every deploy.

https://gorm.io/docs/migration.html

There are strategies we can use for manual migrations, but we would need to consider disabled automigrate.